// custom-emailer/emailer.js
/**
 * Emailer Class
 *
 * @description :: Server-side logic for managing emailers
 *
 */

var emailer = require("nodemailer");
var fs      = require("fs");
var _       = require("underscore");

module.exports = function(options, data) {

	return obj = {

		options: options,

		data: data,

		// Define attachments here
		attachments: [
			{
				fileName: "logoDualconseils.png",
				filePath: "assets/images/logoDualconseils.png",
				cid: "logo@PERP163.fr",
			}
		],

		send: function() {
			var html = obj.getHtml(obj.options.template, obj.data);
			var attachments = obj.getAttachments(html);
			

			var recipient = obj.options.to.email + "<" + obj.options.to.email + ">";

			if(obj.options.to.firstname && obj.options.to.Lastname) {
				recipient = "'" + obj.options.to.name + " " + obj.options.to.surname + "' <" + obj.options.to.email + ">";
			}

			var messageData = {
			  to: recipient,
			  from: "'PERP163.fr'",
			  subject: obj.options.subject,
			  html: html,
			  generateTextFromHTML: true,
			  attachments: attachments
			};

			transport = obj.getTransport();
			transport.sendMail(messageData, function(error, response) {
			    if (error) {
			        console.log(error);
			    } else {
			        //console.log("Message sent: " + response.message);
			    }
			});
		},

		getTransport: function() {
			return emailer.createTransport ("SMTP", {
				service: "Gmail",
				auth: {
					user: "perp163@dualconseils.com",
					pass: "zed88wj99"
				}
			});
		},

		getHtml: function (templateName, data) {
			templatePath = "assets/templates/email/" + templateName + ".ejs";
			templateContent = fs.readFileSync(templatePath, encoding="utf8");
			return _.template(templateContent, data, {interpolate: /\{\{(.+?)\}\}/g});
		},

		getAttachments: function (html) {
			var attachments = [];

			for(var attachment in obj.attachments) {
				attachment = obj.attachments[attachment];
				if (html.search("cid:" + attachment.cid) > -1) {
					attachments.push(attachment);
				}
			}

			return attachments;
		}
	}
};
